### ISSUE TITLE
New release 7.0.0+ not getting a deeplink data on first launch, when app is on resume

### ISSUE THREAD

**Issue by followthemoney1** [RodrigoSMarques/flutter_branch_sdk#265]
**Describe the bug**
When I've updated the plugin to 7.0.0, I do not get new deeplink data or mostly get old deeplink when the app in OnForeground or not killed. The old version before 7.0.0 worked as a charm

**To Reproduce**

1. Integrate branch sdk
2. Run and open the app, swipe to hide an app
3. Open a deeplink, in my case, I've used [https://opk1d.test-app.link/Y82qFg16GEb] with parameters 
![Untitled](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/8526612/4ef3a07d-c5e8-4417-8e88-4be3127ac7bf)
![Untitled](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/8526612/78156682-16f5-4441-a829-7bde663712a0)
4. Logs which I see when apps open from branch deeplink
![Untitled](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/8526612/6d5bfe74-f650-46d3-ba25-b53148af5303)
5. Looks like the link is coming to the native branch SDK but not to the fluter side, and to flutter side listener it's come only after second deeplink come or I will reopen app 

**Expected behavior**
Deeplink will consumed as in version 6.9.0

**Screenshots**
Can send my branch SDK initialization:
![image](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/8526612/39b5ba47-649c-449f-bca6-197e5125c349)
![Untitled](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/8526612/4ee5df1f-1c7c-4933-a543-57229abe7c8c)
![Untitled](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/8526612/f51fb5cf-ba2a-41f5-a4ac-2924e880c74e)

Manifest:
```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="omitted"
    xmlns:tools="http://schemas.android.com/tools">

    <uses-permission android:name="android.permission.READ_PHONE_STATE" />
    <uses-permission android:name="android.permission.RECORD_AUDIO" />
    <uses-permission android:name="android.permission.CAMERA"
        tools:ignore="PermissionImpliesUnsupportedChromeOsHardware" />
    <uses-permission android:name="android.permission.CAMERA2" />
    <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS" />
    <uses-permission android:name="android.permission.BLUETOOTH" />
    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <uses-permission
        android:name="android.permission.READ_PRIVILEGED_PHONE_STATE"
        tools:ignore="ProtectedPermissions" />
    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
    <uses-permission
        android:name="android.permission.BIND_PHONE_ACCOUNT_SUGGESTION_SERVICE"
        tools:ignore="ProtectedPermissions" />
    <!-- Required only if your app needs to access images or photos
     that other apps created. -->
    <uses-permission android:name="android.permission.READ_MEDIA_IMAGES" />
    <uses-permission android:name="android.permission.READ_MEDIA_VIDEO" />
    <uses-permission android:name="android.permission.READ_MEDIA_AUDIO" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"
        android:maxSdkVersion="32" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"
        android:maxSdkVersion="29" />
    <uses-permission android:name="android.permission.ACTIVITY_RECOGNITION" />
    <uses-permission android:name="android.permission.ACCESS_SURFACE_FLINGER"
        tools:ignore="ProtectedPermissions" />

    <queries>
        <!-- If your app opens https URLs -->
        <intent>
            <action android:name="android.intent.action.VIEW" />
            <data android:scheme="https" />
        </intent>
        <!-- If your app makes calls -->
        <intent>
            <action android:name="android.intent.action.DIAL" />
            <data android:scheme="tel" />
        </intent>
        <!-- If your app emails -->
        <intent>
            <action android:name="android.intent.action.SEND" />
            <data android:mimeType="*/*" />
        </intent>
    </queries>

    <uses-sdk tools:overrideLibrary="com.google.firebase.messaging" />

    <uses-permission
        android:name="com.google.android.gms.permission.AD_ID"
        tools:node="remove" />

    <meta-data
        android:name="google_analytics_adid_collection_enabled"
        android:value="false" />
    <meta-data
        android:name="google_analytics_default_allow_ad_personalization_signals"
        android:value="false" />

    <application
        android:name=".App"
        android:icon="@mipmap/ic_launcher"
        android:label="@string/app_name"
        android:exported="false"
        android:requestLegacyExternalStorage="true"
        android:usesCleartextTraffic="true"
        android:networkSecurityConfig="@xml/network_security_config"
        android:clearTaskOnLaunch="true"
        android:theme="@style/PostSplashScreenTheme">
        <activity
            android:name="com.linusu.flutter_web_auth.CallbackActivity"
            android:autoVerify="true"
            android:exported="true">
            <intent-filter android:label="FitBit Connect">
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <data android:scheme="@string/fitbit_callback_url_scheme" />
            </intent-filter>
        </activity>
        <activity
            android:name=".MainActivity"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:exported="true"
            android:fitsSystemWindows="true"
            android:hardwareAccelerated="true"
            android:launchMode="singleTask"
            android:supportsInlineSuggestions="true"
            android:turnScreenOn="true"
            android:autoVerify="true"
            android:windowSoftInputMode="adjustResize">
            <!-- Specifies an Android theme to apply to this Activity as soon as
                       the Android process has started. This theme is visible to the user
                       while the Flutter UI initializes. After that, this theme continues
                       to determine the Window background behind the Flutter UI. -->
            <meta-data
                android:name="io.flutter.embedding.android.NormalTheme"
                android:resource="@style/NormalTheme" />

            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
            <intent-filter android:autoVerify="true">
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <!-- Fallback to support Braze jasper main and enviroment specific deeplink
                     this approach should be implement as main one
                      CHECK IOS -> Info.plist -> CFBundleURLTypes
                      -->
                <data
                    android:host="*"
                    android:scheme="@string/branch_io_uri_scheme" />
                <data
                    android:host="open"
                    android:scheme="@string/branch_io_uri_scheme" />
                <data android:scheme="https" />
            </intent-filter>

            <!-- Branch URI Scheme -->
            <intent-filter android:autoVerify="true">
                <data
                    android:host="open"
                    android:scheme="@string/branch_io_uri_scheme" />
                <data
                    android:host="reminder"
                    android:scheme="@string/branch_io_uri_scheme" />

                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
            </intent-filter>

            <!-- Branch App Links -->
            <intent-filter android:autoVerify="true">
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <data
                    android:host="@string/branch_io_main_domain_url"
                    android:scheme="https" />
                <data
                    android:host="@string/branch_io_main_domain_url_old"
                    android:scheme="https" /> <!--   Old test app link -->
                <!-- example-alternate domain is required for App Links when the Journeys/Web SDK and Deepviews are used inside your website.  -->
                <data
                    android:host="@string/branch_io_alternate_domain_url"
                    android:scheme="https" />
            </intent-filter>
        </activity>

        <meta-data
            android:name="io.branch.sdk.BranchKey"
            android:exported="false"
            android:value="@string/branch_io_live_key" />
        <meta-data
            android:name="io.branch.sdk.BranchKey.test"
            android:exported="false"
            android:value="@string/branch_io_test_key" />
        <meta-data
            android:name="io.branch.sdk.TestMode"
            android:exported="false"
            android:value="@bool/branch_io_is_test_mode" />

        <meta-data
            android:name="flutterEmbedding"
            android:exported="false"
            android:value="2" />
        <meta-data
            android:name="com.google.android.geo.API_KEY"
            android:exported="false"
            android:value="omitted" />

        <service
            android:name=".JasperFirebaseMessaging"
            android:exported="false">
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>

        <uses-library
            android:name="android.test.runner"
            android:required="false" />

        <provider
            android:name="androidx.core.content.FileProvider"
            android:authorities="${applicationId}.fileProvider"
            android:exported="false"
            android:grantUriPermissions="true"
            tools:replace="android:authorities">
            <meta-data
                android:name="android.support.FILE_PROVIDER_PATHS"
                android:resource="@xml/provider_paths" />
        </provider>

        <meta-data
            android:name="io.flutter.embedding.android.EnableImpeller"
            android:value="true" />

    </application>
    <queries>
        <intent>
            <action android:name="android.intent.action.SEND" />
            <data android:mimeType="text/plain" />
        </intent>
    </queries>
</manifest>

```

**Smartphone (Please complete the following information. remove session if not platform):**
 - Flutter: doctor: 
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.6, on macOS 14.1.1 23B81 darwin-arm64, locale en-UA)
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses
[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] IntelliJ IDEA Community Edition (version 2022.3.1)
[✓] VS Code (version 1.84.0)
[✓] Connected device (4 available)
[✓] Network resources

! Doctor found issues in 1 category.

```
 - Package version: 7.0.0 and 7.0.1
 - OS: 
<img width="456" alt="image" src="https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/8526612/013b51a5-7cab-4b7c-88b0-eb3df98285ab">

 - Device: Samsung S10

**Comment by RodrigoSMarques** [comment]
Same problem reported here #264 

I'm analyzing the issue

**Comment by followthemoney1** [comment]
It looks like I've tried to print intent extra before `super.onResume` inside the application
<img width="1284" alt="image" src="https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/8526612/dc791908-eaad-44b5-8e63-730f4f5d5729">

So, to understand the error. In my flutter activity application, I've did overridden `onResume`:
```
    override fun onNewIntent(intent: Intent) {
        Log.e("FlutterBranchSDK-jasper onNewIntent",intent.extras.toString())
        Log.e("FlutterBranchSDK-jasper onNewIntent",intent.getData().toString())
        super.onNewIntent(intent)
    }

    override fun onResume() {
        Log.e("FlutterBranchSDK-jasper onResume",intent.extras.toString())
        Log.e("FlutterBranchSDK-jasper onResume",intent.getData().toString())
        super.onResume()
```

**Comment by RodrigoSMarques** [comment]
Thank you for the informations. I will analyze.

I cannot reproduce the issue in the sample application.

It's working.

**Comment by followthemoney1** [comment]
@RodrigoSMarques so you can try to reproduce to create a short link which I've provided information there, you can create your own link with short parameters but for me, it was a case. Tell me what I can provide also to help you more

**Comment by RodrigoSMarques** [comment]
HI @followthemoney1 

Would it be possible to do a new test?

I received notification that Branch released a new version of the SDK for Android yesterday:
```
Release 5.7.5
Fix a race condition where two quickly queued consecutive init/reInit requests would overwrite uri arguments and return no params.
```

Run `flutter clean` and run your project so I can get this new version of the SDK.

**Comment by followthemoney1** [comment]
@RodrigoSMarques, yes, I will do tomorrow, probably, just give me a commit or branch link

**Comment by RodrigoSMarques** [comment]
> @RodrigoSMarques was you update a 7.0.2 to follow the latest android sdk or do you have some specific commit?

It's not necessary.
The plugin is configured to always use the latest available version (minor and fix) of Branch's Android SDK.

I still haven't been able to reproduce the issue.

Remains under analysis. The difference from version 6 to 7 is the SDK initialization. All other methods remain the same. That's what's strange.

For personal reasons this week I haven't been able to focus on this problem. Next week I will do a deeper analysis.

**Comment by RodrigoSMarques** [comment]
Hi @followthemoney1 

I fixed the bug, but I need help testing the changes in different scenarios.

Can you help me with testing the new version before publishing it on pub.dev?

Update `pubspec.yaml`, according to the example below:

```dart
#flutter_branch_sdk: ^7.0.1
flutter_branch_sdk:
     git:
       url: https://github.com/RodrigoSMarques/flutter_branch_sdk.git
       ref: v7.0.2-dev
```

I'm waiting for the test results.

Thanks.

**Comment by followthemoney1** [comment]
@RodrigoSMarques give me 1-2 days I will come back, just a bit busy. Thanks for waiting

**Comment by rinaldysam-valbury** [comment]
I have the same issue..

**Comment by RodrigoSMarques** [comment]
> I have the same issue..

test the new version, according to these instructions: [https://github.com/RodrigoSMarques/flutter_branch_sdk/issues/265#issuecomment-1844126587](https://github.com/RodrigoSMarques/flutter_branch_sdk/issues/265#issuecomment-1844126587)

**Comment by RodrigoSMarques** [comment]
Fixed in version [7.0.2](https://pub.dev/packages/flutter_branch_sdk)
